src/components/legend/draw.js

```diff
@@ -539,6 +541,11 @@ function computeTextDimensions(g, gd, legendObj, aTitle) {
         // approximation to height offset to center the font
         // to avoid getBoundingClientRect
         if(aTitle === MAIN_TITLE) {
+            if(!isVertical) {
+                // add extra space between legend title and itmes
+                width += constants.titlePad * 2;
```

Comment:
If it is still not enough, this could be increased to be another value/constant.